app/queries/rewards_query.rb

```diff
@@ -0,0 +1,25 @@
+class RewardsQuery
+  def initialize(rewards:, categories:, params: {})
```

Comment:
I think this class is too complicated ;) 
Because what this query needs to do is to return all rewards or rewards filtered by categories. So you could do:
- add simple tests for query object to be able to refactor it without worries
- remove rewards from initialize attributes (always use Reward.all)
- pass only permitted filters
- and then in body of query: prepare query + filter it by category if needed

You can try to refactor it on your own, but if you want to see my proposition: 
<details>
  <summary>Example class</summary>

   
    class RewardsQuery
      def initialize(filters = {})
        @filters = filters
      end
    
      def results
        prepare_query
        filter_by_category
      end
    
      private
    
      def prepare_query
        @results = Reward.all.includes(:category)
      end
    
      def filter_by_category
        @results = @results.where(category_id: @filters[:category_id]) if @filters[:category_id]
      end
    end
    
 And then you can invoke this class with:
```
@rewards = RewardsQuery.new(index_query_params).results.page(params[:page])

def index_query_params
  params.fetch(:filter, {}).permit(:category_id)
end
 ```

 </details>

Reply:
Thank you for this refactoring suggestion and your proposition - it was very helpful 🙂